phase1-coordinator/src/storage/disk.rs

```diff
@@ -41,10 +41,45 @@ pub struct Disk {
     resolver: DiskResolver,
 }
 
-impl Storage for Disk {
+impl Disk {
+    pub fn clear_round_files(&mut self, current_round_height: u64) -> Result<()> {
+        // Let's first fully clear any files in the next round - these will be
+        // verifications and represent the initial challenges.
+        let next_round_dir = self.resolver.round_directory(current_round_height + 1);
+        self.clear_dir_files(next_round_dir.into(), true)?;
+
+        // Now, let's clear all the contributions made on this round.
+        let round_dir = self.resolver.round_directory(current_round_height);
+        self.clear_dir_files(round_dir.into(), false)
+    }
+
+    fn clear_dir_files(&mut self, path: PathBuf, delete_initial_contribution: bool) -> Result<()> {
```

Comment:
I really like this approach. Resetting the round looks like a well defined problem after checking the folder structure.